### PR TITLE
[FEAT] :  Added  a button to pickup the user fav tech stack before spin it up 🚀

### PR DESCRIPTION
**Pull request by Rudrajiii**
## Context :
- Feat #4
- Successfully added a new "Pick Your Stack" button above the Spin Button.
- See Here

  image

## Why ?
- So that users can test their favourite tech stack just before spinning the machine.
  By Doing this users can also test their "LUCK" now.

## Major Changes :
- Files **SelectedTechStack.astro** , **TechStackModl.astro** is added in components folder !! 
- Files **StackOptions.ts** , **techStackModal.ts** is added in scripts folder !!
  
  image

**I also added a small Pop Up Msg When Users Dream Tech stack gets matched with the Spin Result !! You Should Modify The 
Ui Though As i am not good at that :)
And Here is the Final Result 🚀** 

https://github.com/user-attachments/assets/5f588a99-7acf-43a9-b509-0cf2fc3c55a5

### CODE REVIEW

**Comment by Anurag-Kochar-1** [comment]
Hey @Rudrajiii, I’ve merged this into the v2 branch for now. I'll review and address any additional feedback shortly.

**Comment by Rudrajiii** [comment]
@Anurag-Kochar-1 Sure , Waiting for your response :)

**Comment by Anurag-Kochar-1** [comment]
v2 - https://v2.tech-stack-slot-machine.pages.dev/

**Comment by Anurag-Kochar-1** [comment]
@Rudrajiii I guess I need to fix the junky spin animation for mobile and implement some new little ui enhancements.

I was thinking of releasing a version 2 of this project since it has received a decent amount of attention on reddit. Folks have suggested a lot of cool features to add, such as a tech label/summary on hover, more images, and a "hold to spin" option. What do you think?

**Comment by Rudrajiii** [comment]
@Anurag-Kochar-1  yup!! Sounds great and ya we should add more frontend/backend/deployment/database options!! And I think in #2 issue that person is working on the hover stuff .

**Comment by Anurag-Kochar-1** [comment]
Yeah, he is. However, adding a backend would be overkill, Imo. Initially, I thought of this as just a mini frontend project

**Comment by Anurag-Kochar-1** [comment]
I will make a list of the features that people suggested on reddit

**Comment by Rudrajiii** [comment]
@Anurag-Kochar-1 yes listing all the features to implement would be great 🙌

**Comment by Anurag-Kochar-1** [comment]
- label on hover
- more images (and besides js ecosystem too)
- stackpot like jackpot

There were more ideas, but somehow I am not able to find those comments ;(

**Comment by Rudrajiii** [comment]
@Anurag-Kochar-1 cool stuff , i had a think in my mind like if we let users bet some money in 5-10 spins (can be customized by user)  more or like games present in the famous **Slack** website ig!!

and have u changed the winning animation (**BINGO**) when spin result will be matched with the user selected tech stack in the v2 version yet?

**Comment by Anurag-Kochar-1** [comment]
Not yet, I will sit on the version 2 features tomorrow night. I also thought of something similar, like giving users $10 (fake money) by default and charging $1 per spin. At the end, they could do things like follow or retweet to earn more balance.

**Comment by Rudrajiii** [comment]
@Anurag-Kochar-1 yup , sounds great!!

File: src/components/SelectedTechStack.astro [hunk 1]
```diff
@@ -0,0 +1,131 @@
+---
+interface Props {
+  selectedStack: Record<string, string>;
+}
+
+const { selectedStack: initialSelectedStack } = Astro.props;
```
Comment (Anurag-Kochar-1) [inline src/components/SelectedTechStack.astro:6]: const { selectedStack: initialSelectedStack } = Astro.props;

**Comment by Rudrajiii** [comment]
@Anurag-Kochar-1  hey bruh , any update regarding the version-2?

**Pull request merged**